### PR TITLE
Improve sending messages page

### PR DESCRIPTION
**Pull request by bording**
This fixes up a number problems with the page, particuallary around V6-specific stuff showing up on the older pages as well.

I've also added a warning from the upgrade guide about making sure not to use `IMessageSession` or `IEndpointInstance` from inside a handler.

/CC: @indualagarsamy

### CODE REVIEW

File: nservicebus/messaging/send-a-message.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ partial:batch-sends-performance
 
 ## Inside the incoming message processing pipeline
 
-In some cases, messages might need to be sent as part of handling a message such as, inside a regular message handler, a saga, or in some advanced cases as part of extending the message handling pipeline. In this scenario, all the send operations are linked to the incoming message. The send operations take part in the same transaction scope as that of the message handler, thereby ensuring that the send operations rollback if the handling of the message fails at any stage of the message processing pipeline.
+Messages often need to be sent as part of handling an incoming message. When running in a [transaction mode](/nservicebus/transports/transactions.md) that supports it, these send operations take part in the same transaction as that of the message handler, thereby ensuring that the send operations roll back if the handling of the message fails at any stage of the message processing pipeline.
```
Comment (timbussmann) [inline nservicebus/messaging/send-a-message.md]: should we state something about the batching behavior? (no messages going out if the handler fails to complete)

File: nservicebus/messaging/send-a-message_immediate-dispatch_core_[6.0,].partial.md [hunk 2]
```diff
@@ -1,4 +1,11 @@
+The preferred approach is to use the the explicit immediate dispatch API:
 
-The preferred approach is to use the the explicit immediate dispatch API.
+snippet:RequestImmediateDispatch
 
-snippet:RequestImmediateDispatch
\ No newline at end of file
+WARNING: By specifying immediate dispatch, outgoing messages will not be [batched](/nservicebus/messaging/batched-dispatch.md) or enlisted in the current receive transaction even if the transport has support for it.
+
+Suppressing the ambient transaction to have the outgoing message sent immediately is also possible:
```
Comment (timbussmann) [inline nservicebus/messaging/send-a-message_immediate-dispatch_core_[6.0,].partial.md]: I think we no longer want to support that scenario for v6, so we could just remove that section.

File: nservicebus/messaging/send-a-message.md (hunk 1 above)
Comment (indualagarsamy) [inline nservicebus/messaging/send-a-message.md]: @timbussmann - doesn't the following say that?

>thereby ensuring that the send operations roll back if the handling of the message fails at any stage of the message processing pipeline.

How do you think we should clarify it?

File: nservicebus/messaging/send-a-message_immediate-dispatch_core_[6.0,].partial.md (hunk 2 above)
Comment (indualagarsamy) [inline nservicebus/messaging/send-a-message_immediate-dispatch_core_[6.0,].partial.md]: @timbussmann - which one? Immediate dispatch or suppressing the ambient transaction?

File: nservicebus/messaging/send-a-message.md (hunk 1 above)
Comment (bording) [inline nservicebus/messaging/send-a-message.md]: This is already mentioned in the immediate dispatch section further down the page.

**Comment by bording** [comment]
@indualagarsamy @timbussmann Any more feedback on this? If not, I think this is ready to be merged.

**Comment by indualagarsamy** [comment]
@bording - thanks for the cleanup on this page.

**Pull request merged**